xrmocap/model/registrant/smplify.py

```diff
@@ -352,44 +356,104 @@ def __optimize_stage__(self,
         self.call_hook('before_stage', **hook_kwargs)
 
         kwargs = kwargs.copy()
-        parameters = OptimizableParameters()
-        for key, value in optim_param.items():
-            fit_flag = kwargs.pop(f'fit_{key}', True)
-            parameters.add_param(key=key, param=value, fit_param=fit_flag)
-        optimizer = build_optimizer(parameters, self.optimizer)
+
+        # add individual optimizer choice
+        optimizers = {}
+        if 'individual_optimizer' not in self.optimizer:
+            parameters = OptimizableParameters()
+            for key, value in optim_param.items():
+                fit_flag = kwargs.pop(f'fit_{key}', True)
+                parameters.add_param(key=key, param=value, fit_param=fit_flag)
+            optimizers['default_optimizer'] = build_optimizer(
+                parameters, self.optimizer)
+        else:
+            # set an individual optimizer if optimizer config
+            # is given and fit_{key} is True
+            # update with the default optimizer or ignore otherwise
+            # | {key}_opt_config |  fit_{key}  |       optimizer     |
+            # | -----------------| ------------| --------------------|
+            # |       True       |     True    |   {key}_optimizer   |
+            # |       False      |     True    |   default_optimizer |
+            # |       True       |     False   |        ignore       |
+            # |       False      |     False   |        ignore       |
+            self.individual_optimizer = True
+            _optim_param = optim_param.copy()
+            for key in _optim_param.copy().keys():
+                parameters = OptimizableParameters()
+                fit_flag = kwargs.pop(f'fit_{key}', True)
+                if f'{key}_optimizer' in self.optimizer.keys() and fit_flag:
+                    value = _optim_param.pop(key)
+                    parameters.add_param(
+                        key=key, param=value, fit_param=fit_flag)
+                    optimizers[key] = build_optimizer(
+                        parameters, self.optimizer[f'{key}_optimizer'])
+                    self.logger.info(f'Add an individual optimizer for {key}')
+                elif not fit_flag:
+                    _optim_param.pop(key)
+                else:
+                    self.logger.info(f'No optimizer defined for {key}, '
+                                     'get the default optimizer')
+
+            if len(_optim_param) > 0:
+                parameters = OptimizableParameters()
+                if 'default_optimizer' not in self.optimizer:
+                    self.logger.error(
+                        'Individual optimizer mode is selected but '
+                        'some optimizaters are not defined. '
+                        'Please set the default_optimzier or set optimizer '
+                        f'for {_optim_param.keys()}.')
```

Comment:
Always remember raise an error after `self.logger.error()`.
`raise KeyError`

Reply:
Fixed in [cab8731](https://github.com/openxrlab/xrmocap/pull/56/commits/cab87319d90d6f7f6e5f611602f0b4c4d7d6f938).